apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs

```diff
@@ -518,44 +537,57 @@ private void StartAcquisitionTasksIfNecessary()
-                int maxAvailableRequests = maxActiveSessions - liveOrRequestedSessionCount;
+                // How many more sessions do we need to create in order to get the pool to the minimum size, after satisfying
+                // pending callers, assuming that all the in-flight session requests succeed, and there are no more requests?
+                int newSessionsToSatisfyPool = (pendingAcquisitionCount + minPoolSize) - (poolSize + inFlightRequests);
+                // How many more sessions *can* we create without going over the maximum number of active sessions?
+                int maxPossibleNewSessions = maxActiveSessions - liveOrRequestedSessionCount;
+
+                int actualNewSessions = Math.Min(newSessionsToSatisfyPool, maxPossibleNewSessions);
+
+                // Now we need to know how many batch session create requests we need to make.
+                int partialBatchSize = actualNewSessions % createSessionBatchMaxSize;
+                int totalWholeBatches = actualNewSessions / createSessionBatchMaxSize;
 
-                int actualNewRequests = Math.Min(newRequestsToSatisfyPool, maxAvailableRequests);
+                for (int i = 0; i < totalWholeBatches; i++)
+                {
+                    Parent.ConsumeBackgroundTask(PrepareNewSessionsAsync(createSessionBatchMaxSize), "session creation");
+                }
 
-                for (int i = 0; i < actualNewRequests; i++)
+                if(partialBatchSize > 0)
                 {
-                    Parent.ConsumeBackgroundTask(PrepareNewSessionAsync(), "session creation");
+                    Parent.ConsumeBackgroundTask(PrepareNewSessionsAsync(partialBatchSize), "session creation");
                 }
 
-                async Task PrepareNewSessionAsync()
+                async Task PrepareNewSessionsAsync(int batchSize)
                 {
                     try
                     {
-                        var session = await CreatePooledSessionAsync(CancellationToken.None).ConfigureAwait(false);
-                        ReleaseInactiveSession(session, maybeCreateReadWriteTransaction: true);
+                        var sessions = await CreatePooledSessionsAsync(batchSize, CancellationToken.None).ConfigureAwait(false);
+                        foreach (var session in sessions)
+                        {
+                            ReleaseInactiveSession(session, maybeCreateReadWriteTransaction: true);
+                        }
                     }
                     // Note: we expect this to always actually be an RpcException, but we don't want to end up unhealthy
                     // with a lot of 
                     catch (Exception e)
                     {
-                        Parent._logger.Warn(() => $"Failed to create session for {_databaseName}", e);
+                        Parent._logger.Warn(() => $"Failed to batch create sessions for {_databaseName}", e);
```

Comment:
Should this block set _lastUnhealthyBatch?

Reply:
Ah, it looks like that's handled by CreatePooledSessionsAsync. It might be worth mentioning that here in a comment. (It's slightly odd for the logging to be separate from the update. Maybe we should move this log into CreatePooledSessionsAsync.